### PR TITLE
[sqlalchemy] Add table and column comment support

### PR DESCRIPTION
**Pull request by susodapop**
## Description

This PR consolidates the changes from #306 and #308 after review + tests + refactor which I placed into a staging branch to coordinate reviews and testing.

### CODE REVIEW

**Review by cbornet — APPROVED** [review]
LGTM

File: src/databricks/sqlalchemy/test_local/e2e/test_basic.py [hunk 1]
```diff
@@ -463,3 +499,45 @@ def test_get_columns(db_engine, sample_table: str):
...
+        yield tbl
+
+        md.drop_all(bind=engine)
+
+    def test_table_comment_reflection(self, inspector: Inspector, table: Table):
+        tbl_name = table.name
+
+        comment = inspector.get_table_comment(tbl_name)
+
+        assert comment == {"text": "table comment"}
```
Comment (cbornet) [inline src/databricks/sqlalchemy/test_local/e2e/test_basic.py:543]: Could also check column comments with `inspector.get_columns` ?

Comment (susodapop) [reply src/databricks/sqlalchemy/test_local/e2e/test_basic.py:543]: Yes, you already did this with the `test_column_comments`, no? I can rewrite it to use the same fixtures and be more efficient.

Comment (susodapop) [reply src/databricks/sqlalchemy/test_local/e2e/test_basic.py:543]: woof, haven't had my coffee yet. i understand your comment now. I've added an extra method to this test class that simply checks for reflection of the column comment. thank you!

**Review by rcypher-databricks — APPROVED** [review]

File: src/databricks/sqlalchemy/test_local/e2e/test_basic.py (hunk 1 above)
Comment (cbornet) [reply src/databricks/sqlalchemy/test_local/e2e/test_basic.py:540]: Awesome 👍 !

**Pull request merged**